File: app/src/main/java/com/example/chatgptapp/utils/OkhttpUtil.java
package com.example.chatgptapp.utils;

import org.json.JSONArray;
import org.json.JSONObject;

import okhttp3.Callback;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.logging.HttpLoggingInterceptor;

public class OkhttpUtil {
    private static String apiKey = "pk-fCTNqrQIWZTAwypRpqfBCBajxDJPRPMZZLkraUxkxzHcmSJY";
    private String url = "https://api.pawan.krd/v1/chat/completions";
    private static String model = "gpt-3.5-turbo";

    private static double temperature = 0.7;
    private static int maxTokens = 1000;
    private String contentSys = "Good Assistant";
    private String contentUsr = "Who are you?";

    public void ipConfig(Callback newCallback) {
        HttpLoggingInterceptor httpLoggingInterceptor =
                new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BASIC);
        OkHttpClient.Builder builder = new OkHttpClient.Builder()
                .addInterceptor(httpLoggingInterceptor);
        OkHttpClient client = builder.build();

        MediaType ip_mediaType = MediaType.parse("text/plain");
        RequestBody body = RequestBody.create("", ip_mediaType);
        Request ipRequest = new Request.Builder()
                .url("https://api.pawan.krd/resetip")
                .post(body)
                .addHeader("Authorization", "Bearer " + apiKey)
                .build();

        client.newCall(ipRequest).enqueue(newCallback);
    }

    public void doPost(Callback newCallback) {

        try {
            HttpLoggingInterceptor httpLoggingInterceptor =
                    new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BASIC);
            OkHttpClient.Builder builder = new OkHttpClient.Builder()
                    .addInterceptor(httpLoggingInterceptor);
            OkHttpClient client = builder.build();

            MediaType mediaType = MediaType.parse("application/json");

            JSONObject json = new JSONObject();
            json.put("model", model);
            json.put("max_tokens", maxTokens);

            JSONArray messages = new JSONArray();

            JSONObject systemMessage = new JSONObject();
            systemMessage.put("role", "system");
            systemMessage.put("content", contentSys);
            messages.put(systemMessage);

            JSONObject userMessage = new JSONObject();
            userMessage.put("role", "user");
            userMessage.put("content", contentUsr);
            messages.put(userMessage);

            json.put("messages", messages);
            String jsonString = json.toString();

            RequestBody requestBody = RequestBody.create(jsonString, mediaType);

            Request request = new Request.Builder()
                    .url(url)
                    .post(requestBody)
                    .addHeader("Authorization", "Bearer " + apiKey)
                    .addHeader("Content-Type", "application/json")
                    .build();

            client.newCall(request).enqueue(newCallback);

        } catch (Exception e) {
            // Handle exception here
        }
    }

    
    public static String getGptAnswer(String response){
        try {
            JSONObject json = new JSONObject(response);

            JSONArray choices = json.getJSONArray("choices");
            if (choices.length() > 0) {
                JSONObject choice = choices.getJSONObject(0);
                JSONObject message = choice.getJSONObject("message");
                String content = message.getString("content");
                return content;
            }
        }
        catch (Exception e){
            return "暂时没有结果";
        }
        return "";
    } 

    public static void setApiKey(String apiKey) {
        apiKey = apiKey;
    }

    public void setContentUsr(String contentUsr) {
        this.contentUsr = contentUsr;
    }

    public void setContentSys(String contentSys) {
        this.contentSys = contentSys;
    }

    public static void setMaxTokens(int maxTokens) {
        maxTokens = maxTokens;
    }

    public static void setModel(String model) {
        model = model;
    }

    public static void setTemperature(double temperature) {
        temperature = temperature;
    }


}

File: app/src/main/java/com/example/chatgptapp/utils/OkhttpUtil.kt
package com.czl.comm.utils

import com.google.gson.Gson
import okhttp3.*
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.RequestBody.Companion.toRequestBody
import okhttp3.logging.HttpLoggingInterceptor

class OkhttpUtil {
    var firstContent = ""
    var received = ""
    var newContent = ""
    var url = "https://ai.fakeopen.com/api/conversation"

    companion object {
        var apiKey = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Ik1UaEVOVUpHTkVNMVFURTRNMEZCTWpkQ05UZzVNRFUxUlRVd1FVSkRNRU13UmtGRVFrRXpSZyJ9.eyJodHRwczovL2FwaS5vcGVuYWkuY29tL3Byb2ZpbGUiOnsiZW1haWwiOiJjaGVuemlsb25nMjBAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWV9LCJodHRwczovL2FwaS5vcGVuYWkuY29tL2F1dGgiOnsidXNlcl9pZCI6InVzZXItd1puT05TbmZXTGdTazVzbFN1T2JZNFR2In0sImlzcyI6Imh0dHBzOi8vYXV0aDAub3BlbmFpLmNvbS8iLCJzdWIiOiJnb29nbGUtb2F1dGgyfDEwMDM1NjUzODM0MDEwNzY4Mjc0NiIsImF1ZCI6WyJodHRwczovL2FwaS5vcGVuYWkuY29tL3YxIiwiaHR0cHM6Ly9vcGVuYWkub3BlbmFpLmF1dGgwYXBwLmNvbS91c2VyaW5mbyJdLCJpYXQiOjE2ODU2MjUzNTEsImV4cCI6MTY4NjgzNDk1MSwiYXpwIjoiVGRKSWNiZTE2V29USHROOTVueXl3aDVFNHlPbzZJdEciLCJzY29wZSI6Im9wZW5pZCBwcm9maWxlIGVtYWlsIG1vZGVsLnJlYWQgbW9kZWwucmVxdWVzdCBvcmdhbml6YXRpb24ucmVhZCBvcmdhbml6YXRpb24ud3JpdGUifQ.dbW1j-ijPhZS-D-d0hx93TK57TNYLP0--C_AcZ_hQ397m0dHwI7CL5x3uhC4i5RPYFz3xzGgOyi8oR_PsBazSXvvwoVo_-lYzrUeaUz2nlOjYwCQjsE_ylfvn3vJUzYW6c4Kb7PvADVYwo-fDN3h4loBhd8yn5KVquyFVDWdpTVcz2nl30ct8sbuGNd8J_cWzCinbOFB4PPTteQm6PdR_GUNhEtb2aHlA7ECxeAQJ-ge9hugq2TRVfe6xEi3H7Nrbnq9jjrWGxj9suwium51c8NS_jA-q9m937gXNrlVyVDeQdq7kd_OuxDuOX8TXZC3OUBfAEi1XxOdfVHFqRby6w"
    }

    fun doPost(newCallback: Callback) {
        val httpLoggingInterceptor =
            HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BASIC)
        val builder = OkHttpClient.Builder()
            .addInterceptor(httpLoggingInterceptor)
        val client = builder.build()

        val conversations = mutableListOf<Map<String, String>>()

        if (firstContent.isNotEmpty()) {
            conversations.add(mapOf("role" to "user", "content" to firstContent))
        }
        if (received.isNotEmpty()) {
            conversations.add(mapOf("role" to "assistant", "content" to received))
        }
        if (newContent.isNotEmpty()) {
            conversations.add(mapOf("role" to "user", "content" to newContent))
        }
        apiKey = apiKey?.replace(":\"","")
        val requestBody =
            Gson().toJson(mapOf("conversations" to conversations, "accessToken" to apiKey, "apiReverseProxyUrl" to url))
                .toRequestBody("application/json".toMediaType())

        var request = Request.Builder()
            .url(url)
            .post(requestBody)
            .build()
        client.newCall(request).enqueue(newCallback);
    }
}

